### PR TITLE
Fix: CSS not applying on responsive devices and multi-buttons settings not visible in editor issue.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

= 1.25.4 =
* Fix: CSS not applying on responsive devices.
* Fix: Multi Buttons - Child setting are not visible in editor.

### Changes Made
  - Add deviceType attribute in attribute.js
  - Update deviceType attribute withSelect
  - use addBlockEditorDynamicStyles componentDidUpdated
  - Updated css class in styling.js

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Test cases for Responsive CSS
  - Add color or any style to block and check it in tablet/mobile preview.
  - Add a responsive CSS like font size on tablet/mobile and check if it applies in the editor or not.
  - Check style tag is loaded in an iframe when the preview is tablet/mobile.
- Test case for multi-buttons
  - Drag and drop multi-button in master 
  - Click on the button child and check child setting is visible or not.
  - Switch to the above branch and it will get displayed.
- Masonry changes
  - It is already tested and released no need to test.
  
### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-block-helper.php [hunk 1]
```diff
@@ -5619,10 +5619,19 @@ public static function get_gallery_css( $attr, $id ) {
 				$selectors = array();
 				if ( isset( $attr['masonryGutter'] ) && '' !== $attr['masonryGutter'] ) {
 					$selectors = array(
-						'.wp-block-gallery.columns-' . $col_count . ' ul.blocks-gallery-grid' => array(
+						'.wp-block-gallery.has-nested-images.columns-' . $col_count => array(
```
Comment (RajkiranBagal) [inline classes/class-uagb-block-helper.php:5622]: Purpose of this change in this PR?

File: classes/class-uagb-block-helper.php [hunk 2]
```diff
@@ -5653,6 +5662,55 @@ public static function get_gallery_css( $attr, $id ) {
 		public static function get_masonry_gallery_css() {
 
 			$selectors = array(
+				'.wp-block-gallery.has-nested-images'    => array(
+					'display'      => 'block',
+					'column-count' => '3',
```
Comment (RajkiranBagal) [inline classes/class-uagb-block-helper.php:5667]: Why is this change in this PR?

File: classes/class-uagb-block-helper.php [hunk 3]
```diff
@@ -5689,9 +5747,19 @@ public static function get_masonry_gallery_css() {
...
 				),
+				/* End Backword */
 			);
 
 			$m_selectors = array(
+				'.wp-block-gallery[class*="columns-"].blocks-gallery-grid' => array(
+					'column-count' => '2',
+					'column-gap'   => '1em',
+					'display'      => 'unset',
+				),
```
Comment (RajkiranBagal) [inline classes/class-uagb-block-helper.php:5758]: Why is this change in this PR?

File: classes/class-uagb-init-blocks.php [hunk 4]
```diff
@@ -301,8 +301,11 @@ public function editor_assets() {
 				'dependencies' => array(),
 				'version'      => UAGB_VER,
 			);
-		$script_dep      = array_merge( $script_info['dependencies'], array( 'wp-blocks', 'wp-i18n', 'wp-element', 'wp-components', 'wp-editor', 'wp-api-fetch' ) );
-
+		global $pagenow;
+		$script_dep = array_merge( $script_info['dependencies'], array( 'wp-blocks', 'wp-i18n', 'wp-element', 'wp-components', 'wp-api-fetch' ) );
```
Comment (RajkiranBagal) [inline classes/class-uagb-init-blocks.php:305]: Why is this change in this PR?

File: src/blocks/forms/edit.js [hunk 5]
```diff
@@ -1070,17 +1067,21 @@ class UAGBFormsEdit extends Component {
...
 	
 	renderReadyClasses(id){
-		var mainDiv = document.getElementById( "block-" + id )
-		var formscope = mainDiv.getElementsByClassName('uagb-forms__outer-wrap')
+		const iframeEl = document.querySelector( `iframe[name='editor-canvas']` );
+		var mainDiv;
+		var formscope;
+		if( iframeEl ){
+			mainDiv = iframeEl.contentDocument.getElementById( "block-" + id )
+			formscope = mainDiv.getElementsByClassName('uagb-forms__outer-wrap')
```
Comment (RajkiranBagal) [inline src/blocks/forms/edit.js:1080]: Why is this change in this PR?

File: src/blocks/table-of-contents/edit.js [hunk 6]
```diff
@@ -895,8 +884,17 @@ export default compose(
 		}
 
 		var level = 0;
+		const iframeEl = $( `iframe[name='editor-canvas']` ).contents();
+		var mainDiv;
+		if( iframeEl ){
+			mainDiv = iframeEl.find('.is-root-container' ).find('h1, h2, h3, h4, h5, h6' );
```
Comment (RajkiranBagal) [inline src/blocks/table-of-contents/edit.js:890]: Why is this change in this PR?

File: src/blocks/table-of-contents/edit.js [hunk 7]
```diff
@@ -955,7 +953,11 @@ export default compose(
 
 		return {
 			deviceType: deviceType,
-			headers: headers
+			headers: headers,
+			attributes: {
+				...ownProps.attributes,
```
Comment (RajkiranBagal) [inline src/blocks/table-of-contents/edit.js:958]: use props.attributes

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: src/blocks/table-of-contents/edit.js (hunk 6 above)
Comment (AbhijitNage123) [reply src/blocks/table-of-contents/edit.js:890]: To find out heading tag on responsive devices

File: src/blocks/forms/edit.js (hunk 5 above)
Comment (AbhijitNage123) [reply src/blocks/forms/edit.js:1080]: Forms get broken when check on a responsive device due to that I made these changes

File: classes/class-uagb-init-blocks.php (hunk 4 above)
Comment (AbhijitNage123) [reply classes/class-uagb-init-blocks.php:305]: Just updated condition as we discuss with widget area PR. load wp-editor dependency only on editor not widget editor

File: classes/class-uagb-block-helper.php (hunk 1 above)
Comment (AbhijitNage123) [reply classes/class-uagb-block-helper.php:5622]: Actually, I pull of mater in this branch due to which it shows this changes

**Comment by RajkiranBagal** [comment]
@AbhijitNage123 All the changes you have done in this PR should be represented in the title, description and each change should have their test cases in the Pr description. Currently the title, description and testcases are only for the one change.

**Comment by AbhijitNage123** [comment]
Updated the test case in the PR description.

**Comment by RajkiranBagal** [comment]
> Updated the test case in the PR description.
Please update PR title and description accordingly

File: src/blocks/table-of-contents/edit.js (hunk 7 above)
Comment (AbhijitNage123) [reply src/blocks/table-of-contents/edit.js]: Done

File: classes/class-uagb-block-helper.php (hunk 2 above)
Comment (AbhijitNage123) [reply classes/class-uagb-block-helper.php:5667]: Pull master due to which its showing this changes

File: classes/class-uagb-block-helper.php (hunk 3 above)
Comment (AbhijitNage123) [reply classes/class-uagb-block-helper.php:5758]: Pull master in dev due to which its showing this

**Review by RajkiranBagal — APPROVED** [review]

File: readme.txt [hunk 8]
```diff
@@ -169,6 +169,10 @@ The awesome React FontIconPicker is developed by [Alessandro Benoit](http://code
 
 == Changelog ==
 
+= 1.25.4 =
+* Fix: CSS not applying on responsive devices.
+* Fix: Multi Buttons - Child setting are not visible in editor.
```
Comment (sandeshjangam) [inline readme.txt:174]:
```suggestion
* Fix: CSS was not applying on responsive devices.
* Fix: Multi Buttons - Child settings were not visible in the editor.
```

**Review by sandeshjangam — APPROVED** [review]

**Comment by akshaylakh** [comment]
working fine

File: readme.txt [hunk 9]
```diff
@@ -169,6 +169,10 @@ The awesome React FontIconPicker is developed by [Alessandro Benoit](http://code
 
 == Changelog ==
 
+= 1.25.4 =
```
Comment (sandeshjangam) [inline readme.txt:172]:
```suggestion
= 1.25.4 - MONDAY, 7TH FEBRUARY 2022 =
```

**Pull request merged**